libs/providers/ofrep-web/src/lib/ofrep-web-provider.ts

```diff
@@ -194,34 +204,42 @@ export class OFREPWebProvider implements Provider {
       }
 
       if (response.httpStatus !== 200) {
-        handleEvaluationError(response);
+        throw new Error(`Failed OFREP bulk evaluation request, status: ${response.httpStatus}`);
       }
 
       const bulkSuccessResp = response.value;
       const newCache: FlagCache = {};
 
-      bulkSuccessResp.flags?.forEach((evalResp: EvaluationResponse) => {
-        if (isEvaluationFailureResponse(evalResp)) {
-          newCache[evalResp.key] = {
-            errorCode: evalResp.errorCode,
-            errorDetails: evalResp.errorDetails,
-            reason: StandardResolutionReasons.ERROR,
-          };
-        }
+      if ('metadata' in bulkSuccessResp) {
+        this._flagSetMetadataCache = bulkSuccessResp.metadata || {};
```

Comment:
Should we clear metadata every time we call this func if we don't get new metadata? Or just in certain conditions?

Reply:
I'd expect it to be included in every response. It won't be sent over the wire if the etag matches.

Reply:
You'd expect it do, but this is just an HTTP response. I'm talking about edge cases where we get a subsequent payload, but the metadata is missing or invalid. We have to do something in that cases (either keep the old, or delete it). Again, it's an edge case, but we have to make a choice.

Reply:
After more consideration, I moved this to where we set the other auxiliary data (like etag) and update it if it's a proper object, otherwise it gets cleared.